winpr/include/config/wtypes.h.in

```diff
@@ -303,7 +303,9 @@ typedef struct s_LUID
 } LUID, *PLUID;
 
 typedef GUID IID;
+#if !(TARGET_OS_MACCATALYST)
 typedef IID* REFIID;
```

Comment:
Can you give a bit of explanation for this change ?

Reply:
Without this directive, compilation fails on Mac Catalyst with an error similar to:

```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFPlugInCOM.h:21:21: error:
typedef redefinition with different types ('CFUUIDBytes' vs 'IID _' (aka 'struct GUID *'))
typedef CFUUIDBytes REFIID;
```

@akallabeth and I agreed that we would try to wrap the typedef in FreeRDP to avoid it altogether on FreeRDP. I had previously tried to rename REFIID to REF_IID in FreeRDP, but was informed that there is a good reason it's called exactly REFIID, namely that it aims to be identical to a Windows built-in type.

Reply:
@iiordanov I´d prefer cleaning up the includes in client code so that the `swift/OpjectiveC` parts do not clash with the `FreeRDP/C  parts (I think I did not make that clear enough last time, sorry)

Reply:
The client code includes only:

```
#include "ios_freerdp.h"
#include "freerdp/gdi/gdi.h"
```

I don't think I can include anything less on the client side. What shall we do @akallabeth ?

Reply:
you can always move the code parts that need `freerdp/gdi/gdi.h` to their own module (`c|m|h` files) so you don´t need to include them directly.

Reply:
They still need to compile on Mac Catalyst and they won't compile at all unless this #define of REFIID is excluded on Mac Catalyst. Am I missing something @akallabeth ?

Reply:
You only have that problem if you include the definition of `REFIID` (which is not part of any `C` standard includes we use in libfreerdp)
So, you only have the problem where you include `freerdp|winpr` AND `iOS` headers/imports

Reply:
I have no control over what iOS headers / imports are present when I compile an iOS app. Hence this change is mandatory when compiling with Mac Catalyst support on iOS.

Reply:
So what I'm saying is that the iOS headers/imports that bring the Mac Catalyst definition of REFIID are included somewhere inside the Apple SDK - I have no control over whether or not they are included. Basically, it's not "my" problem. The FreeRDP library is unusable for building Mac Catalyst (iOS software running on Mac OS X) binaries without this change.

Reply:
@iiordanov ok, fair enough. I´d still remove this part out then, as client side problems should not spread to core library.
(hint: It is part of some `Objective-C CoreFoundation` import, so it can only be an issue where you have `Objective-C imports` mixed with `FreeRDP C includes`, so proper way is to split these.